src/test/java/io/supertokens/test/jwt/storage/JWTKeyStorageTest.java

```
@@ -0,0 +1,271 @@
+                        while (true) {
+                            numberOfIterations.getAndIncrement();
+                            List<JWTSigningKeyInfo> keysFromStorage = noSQLStorage_1.getJWTSigningKeys_Transaction();
+
+                            JWTSigningKeyInfo matchingKeyInfo = null;
+
+                            for (int j = 0; j < keysFromStorage.size(); j ++) {
+                                if (keysFromStorage.get(j).algorithm.equals(algorithm)) {
+                                    matchingKeyInfo = keysFromStorage.get(j);
+                                    break;
+                                }
+                            }
+
+                            if (numberOfIterations.get() <= 2) {
+                                assert matchingKeyInfo == null;
+                            } else {
+                                assert matchingKeyInfo != null;
+                            }
+
+                            try {
+                                Thread.sleep(700);
+                            } catch (InterruptedException e) {
+                                // Ignore
+                            }
+
+                            try {
+                                if (matchingKeyInfo == null) {
+                                    boolean success = noSQLStorage_1.setJWTSigningKeyInfoIfNoKeyForAlgorithmExists_Transaction(new JWTSymmetricSigningKeyInfo(keyId + "-2", 1000, algorithm, "SomeKeyString"));
+
+                                    if (!success) {
+                                        continue;
+                                    }
+
+                                    endValueOfCon2.set(keyId + "-2");
+                                } else {
+                                    endValueOfCon2.set(matchingKeyInfo.keyId);
+                                }
+
+                                break;
+                            } catch (DuplicateKeyIdException e) {
+                                continue;
```

Throw an error instead of continue

---

Done